app/models/concerns/associated_record.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module AssociatedRecord
```

Comment:
Suggestion not request. Just wondering if this could be more explicit, given the coupling with 2 particular models
```suggestion
module ApplicationAndProfileAssociatedRecord
```